cg/meta/report/api.py

```diff
@@ -43,16 +46,97 @@ def __init__(
         self.scout = scout_api
         self.report_validator = ReportValidator(store)
 
-    def create_delivery_report(self, case_id: str, accept_missing_data: bool = False) -> str:
+    def create_delivery_report(
+        self, case_id: str, analysis_date: datetime, accept_missing_data: bool = False
+    ) -> str:
         """Generate the html contents of a delivery report."""
 
-        delivery_data = self._get_delivery_data(case_id)
+        delivery_data = self._get_delivery_data(case_id, analysis_date)
```

Comment:
I suggest that we name the variables whenever there is more than one to avoid bugs just like you did below